assets/variant-list.js

```diff
@@ -12,6 +11,30 @@ class VariantListRemoveButton extends HTMLElement {
 
 customElements.define('variant-list-remove-button', VariantListRemoveButton);
 
+class VariantListRemoveAllButton extends HTMLElement {
+  constructor() {
+    super();
+    const allVariants = Array.from(document.querySelectorAll('[data-variantid]'));
+    const variantsInCart = allVariants.filter((variant) => parseInt(variant.dataset.cartqty) > 0)
+    if (variantsInCart.length === 0) {
+      this.classList.add('hidden')
+    }
+    const variantList = this.closest('variant-list');
+    const items = {}
+    variantsInCart.forEach((variant) => {
```

Comment:
Could we combine this `forEach` and the `filter` above into 1 step?

``` JS
const items = {};

let hasVariantsInCart = false;

allVariants.forEach((variant) => {
  const cartQty = parseInt(variant.dataset.cartqty);
  if (cartQty > 0) {
    hasVariantsInCart = true;
    items[parseInt(variant.dataset.variantid)] = 0;
  }
});

if (!hasVariantsInCart) {
  this.classList.add('hidden');
}
```

Reply:
Separate steps means that the function would need to loop once through `allVariants` and then loop through `variantsInCart`. Practically, the performance impact wouldn't be noticeable unless the variant list is very large, but would like to avoid if possible.